src/main/java/org/openmrs/isanteplus/performancedata/generator/service/impl/PersonGeneratorService.java

```diff
@@ -6,31 +6,27 @@
 import org.openmrs.isanteplus.performancedata.generator.util.IdUtil;
 import org.openmrs.isanteplus.performancedata.generator.util.RandUtil;
 import org.openmrs.isanteplus.performancedata.generator.util.ParamData;
-import org.openmrs.isanteplus.performancedata.model.AbstractEntity;
 import org.openmrs.isanteplus.performancedata.model.Person;
-import org.openmrs.isanteplus.performancedata.model.connection.Inserter;
+import org.openmrs.isanteplus.performancedata.model.connection.ClinicDataChunk;
 import org.springframework.stereotype.Service;
 
 import java.time.LocalDateTime;
 import java.time.LocalTime;
-import java.util.HashSet;
 import java.util.Optional;
-import java.util.Set;
 import java.util.UUID;
 
 @Service
 public class PersonGeneratorService implements EntityGeneratorService {
 
     @Override
-    public Set<AbstractEntity> generateEntities(long amount, LocalDateTime startDate, Inserter ins) {
-        Set<Person> persons = new HashSet<>();
-
+    public ClinicDataChunk generateEntities(long amount, LocalDateTime startDate,
```

Comment:
I would avoid modification the input parameter "dataChunk" and return it. Alternatively, I suggest to return List<Person> to make it more clean code